lib/engine/game/g_18_mag/game.rb

```diff
@@ -419,12 +419,16 @@ def must_buy_train?(_entity)
         def buy_train(operator, train, price = nil)
           @train_bought = true if train.owner == @depot # No idea if this is what Lonny wants
           cost = price || train.price
+          full_cost = train.price if train.owner == @depot
           if price != :free && train.owner == @depot
             if multiplayer?
               corp = %w[2 4].include?(train.name) ? @ldsteg : @mavag
-              operator.spend(cost / 2, @bank)
-              operator.spend(cost / 2, corp)
-              @log << "#{corp.name} earns #{format_currency(cost / 2)}"
+              bank_cost = full_cost ? (full_cost / 2) - (full_cost - cost) : cost / 2
+              corp_cost = full_cost ? full_cost / 2 : cost / 2
+              puts("here, #{bank_cost} #{full_cost} #{bank_cost + full_cost}")
```

Comment:
code could be cleaner. 
When using the kalman kando supporter, there's a train discount. However the corp still gets the half of the full cost of the train. This code handles that case